### ISSUE TITLE
View Page Source URL retains web url instead of view-source:<URL>

### ISSUE THREAD

**Issue by srirambv** [brave/browser-laptop#3734]
**Did you search for similar issues before submitting this one?**

**Describe the issue you encountered:**
View Page Source URL retains full url instead of view-source:<URL>

**Expected behavior:**
When view page source is selected the URL should retain prefix `view-source:` 

- Platform (Win7, 8, 10? macOS? Linux distro?): 
Windows 10 x64

- Brave Version:
0.12.0 RC1

- Steps to reproduce:
    1. Press Ctrl+Alt+U on any open page
    2. New tab opens with the page source with page URL as it is instead of prefixing it with `view-source:` 

- Screenshot if needed:

- Any related issues:
#3685 

cc: @Sh1d0w

**Comment by Sh1d0w** [comment]
While I was working on another issue I digged into this. Turned out that
the url was correctly prefixed with view-source prefix, but when the web
contents load they send `did-navigate` event. The listener for this event
gets the url from it and updates the url bar url value.

Seems like the url sent in that event is always the website url, no matter
if you send it prefixed with `view-source`.

I am not sure where this needs to be fixed - on electron side or on app's
side?

@bbondy any thoughts
